servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/P2CLoadBalancerBuilder.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright © 2023 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.loadbalancer;
+
+import io.servicetalk.client.api.LoadBalancedConnection;
+
+public interface P2CLoadBalancerBuilder<ResolvedAddress, C extends LoadBalancedConnection> {
```

Comment:
💯  agreed, this is the least preferred approach. The main reason is that every `LoadBalancerBuilder` will have to also have a ServiceLoaded provider. With growing number of providers, it makes it harder to control the behavior that way.